### DISCUSSION TITLE
Add `--ignore-missing` option to `nx run` to run non-existent targets as noop without failing

### DISCUSSION THREAD

**Discussion by akwodkiewicz** [nrwl/nx#28415] [Feature Requests]
I'd like to be able to run

```sh
nx run --ignore-missing foobar:some-target 
```

so that Nx will not fail when `some-target` does not exist on foobar, but just do nothing.

## Why?

Obviously I don't want to run `foobar:some-target` verbatim.

In my Dockerfile, I have the project parametrized (`$NX_PROJECT_NAME`) and I am reusing the same Dockerfile with multiple projects.

I want to add an additional target that needs to be run during the image build process, but the target will only be specified for _some_ of the projects that use that Dockerfile.

I'd like to add the following line and call it a day:

```Dockerfile
RUN nx run --ignore-missing $NX_PROJECT_NAME:some-target
```

However, since there is no flag like this one, I need to parse project configuration which feels very roundabout:

```Dockerfile
RUN [[ $(nx show project --json $NX_PROJECT_NAME | jq  '.targets | has("some-target")') == 'true' ]] \
  && nx run $NX_PROJECT_NAME:some-target
```

**Comment by akwodkiewicz** [comment]
Any combination of composing targets differently with `dependsOn` does not work for me, because I want this command to be run at a particular point of time.

**Comment by quezak** [comment]
As a workaround, `run-many` automatically ignores projects for which the target is not defined, and you can pass just one project to `-p`:
```
RUN nx run-many -t some-target -p $NX_PROJECT_NAME
```
